src/telemetry/step_stats.rs

```diff
@@ -0,0 +1,86 @@
+//!
+//! Export metrics collected during protocol run in CSV format. Metrics are partitioned by step.
+
+use crate::telemetry::labels;
+use crate::telemetry::metrics::{INDEXED_PRSS_GENERATED, RECORDS_SENT, SEQUENTIAL_PRSS_GENERATED};
+use crate::telemetry::stats::Metrics;
+use std::collections::{BTreeMap, HashMap};
+use std::io;
+use std::io::{Error, Write};
+
+pub trait CsvExporter {
+    /// Writes the serialized version of this instance into the provided writer in CSV format.
+    ///
+    /// ## Errors
+    /// Returns an error if an IO error occurs while writing to `W`.
+    fn export<W: io::Write>(&self, w: &mut W) -> Result<(), io::Error>;
+}
+
+impl CsvExporter for Metrics {
+    fn export<W: Write>(&self, w: &mut W) -> Result<(), Error> {
+        // first thing is to collect all the steps and metrics emitted
+        let mut steps_stats = StepsStats::default();
+        for (counter_name, details) in &self.counters {
+            if let Some(steps) = details.dimensions.get(labels::STEP) {
+                for (step, val) in steps {
+                    steps_stats.offer(step, counter_name.as_str(), *val);
+                }
+            }
+        }
+
+        // then dump them to the provided Write interface
+        writeln!(w, "Step,Records sent,Indexed PRSS,Sequential PRSS")?;
```

Comment:
this does not provide the breakdown per helper. Except prss, everything else is symmetrical (equal number of records sent by each helper). I can add that breakdown if required

Reply:
It would be useful to have the per-helper breakdown.  With sparse multiplication, the work involved can be asymmetric and understanding that might be helpful.

It is more stuff, but post-processing should be easy.

```suggestion
        writeln!(w, "Step,Records Sent,Indexed PRSS,Sequential PRSS")?;
```

Reply:
Would need to work on that - added a todo for now. The problem is `Metrics` struct does not have this breakdown :( It can tell the total value and value per single dimension